src/field_theory/splitting_field.lean

```diff
@@ -92,63 +93,409 @@ theorem nat_degree_remove_factor' {f : K[X]} {n : ℕ} (hfn : f.nat_degree = n+1
   f.remove_factor.nat_degree = n :=
 by rw [nat_degree_remove_factor, hfn, n.add_sub_cancel]
 
-/-- Auxiliary construction to a splitting field of a polynomial, which removes
-`n` (arbitrarily-chosen) factors.
-
-Uses recursion on the degree. For better definitional behaviour, structures
-including `splitting_field_aux` (such as instances) should be defined using
-this recursion in each field, rather than defining the whole tuple through
-recursion.
--/
-def splitting_field_aux (n : ℕ) : Π {K : Type u} [field K], by exactI Π (f : K[X]), Type u :=
-nat.rec_on n (λ K _ _, K) $ λ n ih K _ f, by exactI
-ih f.remove_factor
+/-- Auxiliary construction to a splitting field of a polynomial. Uses induction on the degree. -/
+def splitting_field_aux : Π (n : ℕ) {K : Type u} [field K], by exactI Π (f : K[X]), Type u
+| 0 K fK f := K
+| (n + 1) K fK f := by exactI splitting_field_aux n f.remove_factor
```

Comment:
Is there a reason for removing this docstring?

Reply:
just accident!